cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexManagerImpl.java

```diff
@@ -256,6 +257,34 @@ public String ensureIndexed(List<FieldSort> fieldNames,
                                 String indexName,
                                 IndexType indexType,
                                 String tokenize) throws QueryException {
+
+        // If the index name is null, a name should be generated, but first we should check
+        // if an index already exists that matches the request index definition.
+        if (indexName == null) {
+            List<Index> indexes = this.listIndexes();
+            // Create a copy of the field names list so we know its mutable.
+            List<FieldSort> mFieldNames = new ArrayList<FieldSort>(fieldNames);
+            Collections.sort(mFieldNames);
```

Comment:
I'm not too keen on this copy and sort. Also how do we know that `index.fieldNames` (inside the for loop) doesn't need to be copied before sorting as well?

I would prefer a different strategy such as:
- put them both into a `Set` and use `Set.equals()`
- don't sort them but use `a.containsAll(b) && a.size() == b.size()`